stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt

```diff
@@ -197,43 +234,57 @@ internal fun BoxScope.DefaultMessagesHelperContent(
 
     val offset = messagesState.focusedMessageOffset.collectAsState()
 
-    LaunchedEffect(
-        newMessageState,
-        firstVisibleItemIndex,
-        focusedItemIndex,
-        offset.value
-    ) {
-        if (focusedItemIndex != -1 && !lazyListState.isScrollInProgress) {
+    LaunchedEffect(newMessageState, focusedItemIndex, offset.value, scrollToStartState) {
+
+        if (focusedItemIndex != -1 &&
+            !lazyListState.isScrollInProgress &&
+            scrollToStartState is ScrollToFocusedMessage
+        ) {
             coroutineScope.launch {
                 lazyListState.scrollToItem(focusedItemIndex, offset.value ?: 0)
             }
         }
 
+        if (scrollToStartState == ScrollToNewestMessages) {
+            lazyListState.scrollToItem(0)
+        }
+
         when {
-            !lazyListState.isScrollInProgress && newMessageState == Other &&
-                firstVisibleItemIndex < 3 -> coroutineScope.launch {
-                lazyListState.animateScrollToItem(0)
+            focusedItemIndex == -1 &&
+                !lazyListState.isScrollInProgress &&
+                newMessageState == Other &&
+                scrollToStartState is Idle &&
+                areNewestMessagesLoaded &&
+                firstVisibleItemIndex < 3 -> {
+                coroutineScope.launch {
+                    lazyListState.animateScrollToItem(0)
+                }
             }
 
-            !lazyListState.isScrollInProgress && newMessageState == MyOwn -> coroutineScope.launch {
-                if (firstVisibleItemIndex > 5) {
-                    lazyListState.scrollToItem(5)
+            focusedItemIndex == -1 &&
+                !lazyListState.isScrollInProgress &&
+                newMessageState == MyOwn &&
+                areNewestMessagesLoaded &&
+                scrollToStartState is Idle -> {
+                coroutineScope.launch {
+                    if (firstVisibleItemIndex > 5) {
+                        lazyListState.scrollToItem(5)
+                    }
+                    lazyListState.animateScrollToItem(0)
                 }
-                lazyListState.animateScrollToItem(0)
             }
         }
     }
 
-    if (abs(firstVisibleItemIndex) >= 3) {
+    if ((messagesState.parentMessageId == null && (abs(firstVisibleItemIndex) >= 3 || !areNewestMessagesLoaded)) ||
+        (messagesState.parentMessageId != null && abs(firstVisibleItemIndex) >= 3)
```

Comment:
Kind of same here - maybe 2 values and we can have `firstValue || secondValue` (obviously not htose names, something meaningful

Reply:
Also to mention - having thread and messages state in the same UI, we might want to split it up in the future